### PR TITLE
[PREVIEW] SL-1791 See all sessions for a listing request

### PR DESCRIPTION
**Pull request by filipczyks**
**Before creating a pull request make sure that:**

https://tools.hmcts.net/jira/browse/SL-1791

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SL-1791

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/model/response/HearingWithSessionsResponse.java [hunk 1]
```diff
@@ -43,6 +45,7 @@ public HearingWithSessionsResponse(Hearing hearing) {
         this.reservedToJudge = hearing.getReservedJudge() != null ? hearing.getReservedJudge().getName() : null;
         this.sessions = hearing.getHearingParts()
             .stream()
+            .sorted(comparing(hearingPart -> hearingPart.getSession().getStart()))
             .map(h -> h.getSession() != null ? new ViewSessionResponse(h.getSession()) : null)
             .filter(Objects::nonNull)
```
Comment (leszekgu) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/model/response/HearingWithSessionsResponse.java]: Can I suggest to do the filter as first thing,
then sorting
then mapping with ViewSessionResponse::new

that would reduce the null number of null checks

Comment (filipczyks) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlevents/model/response/HearingWithSessionsResponse.java]: sure, I don't know why I wrote it that way first time

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/model/response/HearingWithSessionsResponse.java [hunk 2]
```diff
@@ -43,8 +45,9 @@ public HearingWithSessionsResponse(Hearing hearing) {
         this.reservedToJudge = hearing.getReservedJudge() != null ? hearing.getReservedJudge().getName() : null;
         this.sessions = hearing.getHearingParts()
             .stream()
-            .map(h -> h.getSession() != null ? new ViewSessionResponse(h.getSession()) : null)
-            .filter(Objects::nonNull)
+            .filter(h -> h.getSession() != null)
```
Comment (zielinskikamil) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/model/response/HearingWithSessionsResponse.java]: I don't want to be picky but instead h you could use hp :)

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/model/response/HearingWithSessionsResponse.java [hunk 3]
```diff
@@ -43,8 +45,9 @@ public HearingWithSessionsResponse(Hearing hearing) {
...
+            .map(h -> new ViewSessionResponse(h.getSession()))
+            .sorted(comparing(sessionResponse -> sessionResponse.getStart()))
```
Comment (zielinskikamil) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/model/response/HearingWithSessionsResponse.java]: btw, can we make this sorting on the database?

**Pull request merged**